messages/models.tsp

```diff
@@ -3,27 +3,35 @@
  * Edits made directly to this file will be lost.
  */
 
+import "../assistants";
+
 using TypeSpec.OpenAPI;
 
 namespace OpenAI;
 
 model CreateMessageRequest {
-  /** The role of the entity that is creating the message. Currently only `user` is supported. */
-  role: "user";
+  /**
+   * The role of the entity that is creating the message. Allowed values include:
+   * - `user`: Indicates the message is sent by an actual user and should be used in most cases to
+   * represent user-generated messages.
+   * - `assistant`: Indicates the message is generated by the assistant. Use this value to insert
+   * messages from the assistant into the conversation.
+   */
+  role: "user" | "assistant";
 
   /** The content of the message. */
   @minLength(1)
-  @maxLength(32768)
+  @maxLength(256000)
   content: string;
 
-  /**
-   * A list of [File](/docs/api-reference/files) IDs that the message should use. There can be a
-   * maximum of 10 files attached to a message. Useful for tools like `retrieval` and
-   * `code_interpreter` that can access and use files.
-   */
-  @minItems(1)
-  @maxItems(10)
-  file_ids?: string[] = [];
+  /** A list of files attached to the message, and the tools they should be added to. */
+  attachments?: {
+    /** The ID of the file to attach to the message. */
+    file_id?: string;
+
+    /** The tools to add this file to. */
+    tools?: (AssistantToolsCode | AssistantToolsFileSearch)[];
```

Comment:
```suggestion
  attachments?: {
    /** The ID of the file to attach to the message. */
    file_id: string;

    /** The tools to add this file to. */
    tools: (AssistantToolsCode | AssistantToolsFileSearch)[];
```

![image](https://github.com/joseharriaga/openai-in-typespec/assets/21109913/f0f8c485-e6b4-4b9c-94e1-be77a9e319ed)

Reply:
Good find, let me see why it pulled in that nullable incorrectly

Reply:
This may have changed (I hadn't updated the root `openapi.yaml` from the snapshot link); it looks like there are no `required` constraints and the latest generated .tsp matches the doc:

```yaml
        attachments:
          type: array
          items:
            type: object
            properties:
              file_id:
                type: string
                description: The ID of the file to attach to the message.
              tools:
                description: The tools to add this file to.
                type: array
                items:
                  oneOf:
                    - $ref: "#/components/schemas/AssistantToolsCode"
                    - $ref: "#/components/schemas/AssistantToolsFileSearch"
                  x-oaiExpandable: true
          description: A list of files attached to the message, and the tools they were added to.
          nullable: true
```

```
// Tool generated type. Extracts CreateMessageRequest.attachments
model CreateMessageRequestAttachments
  is {
    /** The ID of the file to attach to the message. */
    file_id?: string;

    /** The tools to add this file to. */
    @extension("x-oaiExpandable", true)
    tools?: (AssistantToolsCode | AssistantToolsFileSearch)[];
  }[];
```

```
  /** A list of files attached to the message, and the tools they were added to. */
  attachments: MessageObjectAttachments | null;
```